File: mobile_app/lib/services/auth_api.dart
// lib/services/auth_api.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class AuthApi {
  static const String baseUrl = 'http://10.0.2.2:3000';

  // ---------- REGISTER ---------- (unchanged)
  static Future<void> register({
    required String name,
    required String email,
    required String password,
    required String role,
    required String disabilityType,
    String? grade,
  }) async {
    final url = Uri.parse('$baseUrl/api/auth/register');

    final Map<String, dynamic> body = {
      'name': name,
      'email': email,
      'password': password,
      'role': role,
      'disabilityType': disabilityType,
    };

    if (grade != null && grade.isNotEmpty) {
      body['grade'] = grade;
    }

    final response = await http.post(
      url,
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(body),
    );

    if (response.statusCode != 201) {
      throw Exception("Registration failed: ${response.body}");
    }
  }

  // ---------- LOGIN ----------  ✅ fixed
  static Future<Map<String, dynamic>> login({
    required String email,
    required String password,
  }) async {
    final url = Uri.parse('$baseUrl/api/auth/login');

    final response = await http.post(
      url,
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode({'email': email, 'password': password}),
    );

    final json = jsonDecode(response.body);

  return {
    'token': json['data']['token'],
    'user': json['data']['user'],
  };
}

}

File: test-fe/test-fe/src/services/authApi.ts
const API_BASE =
  import.meta.env.VITE_API_BASE_URL || "http://localhost:3000";

export async function loginUser(email: string, password: string) {
  const res = await fetch(`${API_BASE}/api/auth/login`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ email, password }),
  });

  const json = await res.json();

  if (!res.ok) {
    throw new Error(json?.message || "Login failed");
  }

  // Support both:
  // 1) { token, user }
  // 2) { data: { token, user } }
  const payload = json?.data ?? json;

  if (!payload?.token || !payload?.user) {
    throw new Error("Login response missing token/user");
  }

  return payload; // always { token, user }
}


/**
 * REGISTER
 */
export async function registerUser(payload: {
  name: string;
  email: string;
  password: string;
  role: "student" | "teacher" | "parent";
  disabilityType?: string;
  grade?: number;
  age?: number;
}) {
  const res = await fetch(`${API_BASE}/auth/register`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });

  const data = await res.json();

  if (!res.ok) {
    throw new Error(data.message || "Register failed");
  }

  return data;
}
